amp/WEB-INF/src/org/digijava/kernel/ampapi/endpoints/activity/field/CachingFieldsEnumeratorFactory.java

```diff
@@ -32,7 +32,12 @@ public CachingFieldsEnumerator getDefaultEnumerator() {
     }
 
     public CachingFieldsEnumerator getEnumerator(Long templateId) {
-        return cache.putIfAbsent(templateId, buildEnumerator(new FMVisibility(templateId)));
+        CachingFieldsEnumerator cacheFieldsEnumerator = buildEnumerator(new FMVisibility(templateId));
```

Comment:
@lilitk this change fixes the random error you where geting @ginchauspe was not able to reproduce

Reply:
I was able :)